tests/test_treasury_module.py

```diff
@@ -94,6 +105,179 @@ def test_bundle_creation_with_riskpool_wallet_not_set(
+        instance,
+        owner,
+        testCoin,
+        productOwner,
+        oracleProvider,
+        riskpoolKeeper,
+        capitalOwner,
+        True
+    )
+
+    # prepare allowance too small for riskpool funding 
+    safetyFactor = 2
+    amount = 10000
+    testCoin.transfer(capitalOwner, safetyFactor * amount, {'from': owner})
+    testCoin.approve(instance.getTreasury(), 0.9 * amount, {'from': capitalOwner})
+
+    # ensures bundle creation fails due to insufficient allowance
+    with brownie.reverts("ERC20: insufficient allowance"):
+        gifRiskpool.getContract().createBundle(
+                applicationFilter, 
+                amount, 
+                {'from': capitalOwner})
+
+
+def test_bundle_withdrawal_allowance_too_small(
+    instance: GifInstance,
+    owner: Account,
+    testCoin,
+    productOwner: Account,
+    oracleProvider: Account,
+    riskpoolKeeper: Account,
+    capitalOwner: Account,
+):  
+    applicationFilter = bytes(0)
+
+    # prepare product and riskpool
+    (gifProduct, gifRiskpool) = getProductAndRiskpool(
+        instance,
+        owner,
+        testCoin,
+        productOwner,
+        oracleProvider,
+        riskpoolKeeper,
+        capitalOwner,
+        True
+    )
+
+    # fund bundle
+    safetyFactor = 2
+    amount = 10000
+    testCoin.transfer(capitalOwner, safetyFactor * amount, {'from': owner})
+    testCoin.approve(instance.getTreasury(), amount, {'from': capitalOwner})
+    riskpool = gifRiskpool.getContract()
+
+    riskpool.createBundle(
+                applicationFilter, 
+                amount, 
+                {'from': capitalOwner})
+
+    bundle = riskpool.getBundle(0)
+    print(bundle)
+
+    (bundleId) = bundle[0]
+
+    # check bundle values with expectation
+    assert bundleId == 1
+    
+    # close bundle
+    riskpool.closeBundle(bundleId, {'from': capitalOwner})
+
+    # prepare allowance that is too small for bundle withdrawal
+    testCoin.approve(instance.getTreasury(), 0.9 * amount, {'from': capitalOwner})
+
+    # ensures that burning bundle fails during token withdrawal due to insufficient allowance
+    with brownie.reverts("ERC20: insufficient allowance"):
+        riskpool.burnBundle(
+                bundleId, 
+                {'from': capitalOwner})
+
+
+def test_payout_allowance_too_small(
+    instance: GifInstance,
+    owner: Account,
+    testCoin,
+    productOwner: Account,
+    oracleProvider: Account,
+    riskpoolKeeper: Account,
+    capitalOwner: Account,
+    customer: Account,
+):  
+    applicationFilter = bytes(0)
+
+    (gifProduct, gifRiskpool) = getProductAndRiskpool(
+        instance,
+        owner,
+        testCoin,
+        productOwner,
+        oracleProvider,
+        riskpoolKeeper,
+        capitalOwner,
+        True
+    )
+
+    # prepare product and riskpool
+    safetyFactor = 2
+    amount = 10000
+    testCoin.transfer(capitalOwner, safetyFactor * amount, {'from': owner})

```

Comment:
please check if capitalOwner represents the riskpool wallet, if this is the case change test case to riskpoolKeeper as investor

Reply:
done